docs/guides/plugins.md

```diff
@@ -1,59 +1,336 @@
+videojs.registerPlugin('examplePlugin', examplePlugin);
 ```
 
-From this point on, your plugin will be added to the Video.js prototype and will show up as a property on every instance created. Make sure you choose a unique name that doesn't clash with any of the properties already in Video.js. Which leads us to...
+The only stipulation with the name of the plugin is that it cannot conflict with any existing player method. After that, any player will automatically have an `examplePlugin` method on its prototype!
+
+## Writing a Class-Based Plugin
+
+As of Video.js 6, there is an additional type of plugin supported: class-based plugins.
+
+At any time, you may want to refer to the [Plugin API docs][api-plugin] for more detail.
+
+### Write a JavaScript Class/Constructor
 
-Step 3: Using A Plugin
-----------------------
-There are two ways to initialize a plugin. If you're creating your video tag dynamically, you can specify the plugins you'd like to initialize with it and any options you want to pass to them:
+If you're familiar with creating [components](components.md), this process is similar. A class-based plugin starts with a JavaScript class (a.k.a. a constructor function).
+
+This can be achieved with ES6 classes:
 
 ```js
-    videojs('vidId', {
-      plugins: {
-        examplePlugin: {
-          exampleOption: true
-        }
-      }
+const Plugin = videojs.getPlugin('plugin');
+
+class ExamplePlugin extends Plugin {
+
+  constructor(player, options) {
+    super(player);
+
+    if (options.customClass) {
+      player.addClass(options.customClass);
+    }
+
+    player.on('playing', function() {
+      videojs.log('playback began!');
     });
+  }
+}
 ```
 
-If you've already initialized your video tag, you can activate a plugin at any time by calling its setup function directly:
+Or with ES5:
 
 ```js
-    var video = videojs('cool-vid');
-    video.examplePlugin({ exampleOption: true });
+var Plugin = videojs.getPlugin('plugin');
+
+var ExamplePlugin = videojs.extend(Plugin, {
+
+  constructor: function(player, options) {
+    Plugin.call(this, player, options);
+
+    if (options.customClass) {
+      player.addClass(options.customClass);
+    }
+
+    player.on('playing', function() {
+      videojs.log('playback began!');
+    });
+  }
+});
+```
+
+For now, this example class-based plugin does the exact same thing as the basic plugin described above - not to worry, we will make it more interesting as we continue!
+
+### Register a Class-Based Plugin
+
+The registration process for class-based plugins is identical to [the process for basic plugins](#register-a-basic-plugin).
+
+```js
+videojs.registerPlugin('examplePlugin', ExamplePlugin);
+```
+
+> **Note:** Because ES6 classes are syntactic sugar on top of existing constructor function and prototype architecture in JavaScript, in all cases `registerPlugin`'s second argument is a function.
+
+### Key Differences from Basic Plugins
```

Comment:
I think people are familiar with basic plugins and, if they are, will have certain expectations as to how they work. These could be subtle differences, so I wanted to call them out here.